app/src/main/java/org/oppia/android/app/translation/AppLanguageWatcherMixin.kt

```diff
@@ -62,8 +66,21 @@ class AppLanguageWatcherMixin @Inject constructor(
     }
 
     // TODO(#52): Hook this up properly to profiles, and handle the non-profile activity cases.
-    val profileId = ProfileId.getDefaultInstance()
-    val appLanguageLocaleDataProvider = translationController.getAppLanguageLocale(profileId)
+    var currentUserProfileId = profileManagementController.getCurrentProfileId()
+
+    if (currentUserProfileId.internalId == -1) {
```

Comment:
Why do we need to check this? Shouldn't ``shouldUseSystemLanguage`` cover all the cases that we care about?